### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by kkraune**

### CODE REVIEW

File: album-recommendation-java/src/main/java/com/mydomain/example/Rot13DocumentProcessor.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package com.mydomain.example;
+
+import com.yahoo.docproc.DocumentProcessor;
+import com.yahoo.docproc.Processing;
+import com.yahoo.document.Document;
+import com.yahoo.document.DocumentOperation;
+import com.yahoo.document.DocumentPut;
+import com.yahoo.document.datatypes.StringFieldValue;
+
+public class Rot13DocumentProcessor extends DocumentProcessor {
```
Comment (jonmv) [inline album-recommendation-java/src/main/java/com/mydomain/example/Rot13DocumentProcessor.java]: This probably makes it a bit hard to search for things you've fed. Should we use something else? 

Ah, it's not plugged in either. Intentionally?

File: album-recommendation-java/src/main/java/com/mydomain/example/MetalSearcher.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Override
+    public Result search(Query query, Execution execution) {
+        QueryTree tree = query.getModel().getQueryTree();
+        if (isMetalQuery(tree)) {
+            OrItem orItem = new OrItem();
+            orItem.addItem(tree.getRoot());
+            orItem.addItem(new WordItem("metal", "album"));
+            tree.setRoot(orItem);
+            query.trace("Metal added", true, 2);
```
Comment (jonmv) [inline album-recommendation-java/src/main/java/com/mydomain/example/MetalSearcher.java]: Nice example.

File: album-recommendation-java/src/test/java/com/mydomain/example/FeedAndSearchSystemTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                     + "    }\n"
+                     + "}\n";
+
+        var yql = "SELECT * FROM SOURCES * WHERE album CONTAINS \"Got to be there\";";
+
+        var deleteResult = endpoint.send(endpoint.request(documentPath).DELETE());
+        assertEquals(200, deleteResult.statusCode());
+
+        // the first query needs a higher timeout than the default 500ms, to warm up the code
+        var emptyResult = endpoint.send(endpoint.request("/search/", Map.of("yql", yql, "timeout", "5s")));
```
Comment (jonmv) [inline album-recommendation-java/src/test/java/com/mydomain/example/FeedAndSearchSystemTest.java]:
```suggestion
        var emptyResult = endpoint.send(endpoint.request("/search/", Map.of("yql", yql,
                                                                            "timeout", "5s")));
```

File: album-recommendation-java/src/test/java/com/mydomain/example/FeedAndSearchSystemTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        assertEquals(200, feedResult.statusCode());
+
+        var searchResult = endpoint.send(endpoint.request("/search/", Map.of("yql", yql)));
+        var searchInspector = new JsonDecoder().decode(new Slime(), searchResult.body().getBytes(UTF_8)).get();
+        assertEquals(200, searchResult.statusCode());
+        assertTrue(searchInspector.field("root").field("fields").field("totalCount").valid());
+        assertEquals(1, searchInspector.field("root").field("fields").field("totalCount").asLong());
+    }
+
+}
```
Comment (jonmv) [inline album-recommendation-java/src/test/java/com/mydomain/example/FeedAndSearchSystemTest.java]: Would be great it if this test tested Rot13 and the searcher!

Comment (jonmv) [reply album-recommendation-java/src/test/java/com/mydomain/example/FeedAndSearchSystemTest.java]: Also, if it used the static data in the `.json` files.

File: album-recommendation-java/src/main/java/com/mydomain/example/Rot13DocumentProcessor.java (hunk 1 above)
Comment (kkraune) [reply album-recommendation-java/src/main/java/com/mydomain/example/Rot13DocumentProcessor.java]: this is something I just copied from elsewhere - the plan is to rewrite it to do something useful. This is very out of context today. same as in basic-search-java so not worse than today - will leave it for now

File: album-recommendation-java/src/test/java/com/mydomain/example/FeedAndSearchSystemTest.java (hunk 4 above)
Comment (kkraune) [reply album-recommendation-java/src/test/java/com/mydomain/example/FeedAndSearchSystemTest.java]: agreed to both. I promise to do more work on this. some time

**Pull request merged**